### PR TITLE
Add `/api/v1/status/flags` API

### PR DESCRIPTION
**Pull request by hi-rustin**
#### What this PR does
Add `/api/v1/status/flags` API and add a test for it.
#### Which issue(s) this PR fixes or relates to

close https://github.com/grafana/mimir/issues/3323

#### Checklist

- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/api/api.go [hunk 1]
```diff
@@ -71,6 +71,10 @@ type Config struct {
 	// initialized, the custom config handler will be used instead of
 	// DefaultConfigHandler.
 	CustomConfigHandler ConfigHandler `yaml:"-"`
+
+	// The values from the flags that have been set.
+	// Used for `/api/v1/status/flags` API.
+	Flags map[string]string `yaml:"-"`
```
Comment (hi-rustin) [inline pkg/api/api.go:77]: I'm not sure does this place is a good place to add it. If you have any suggestions, please help me to find a better place! Thanks!

**Comment by hi-rustin** [comment]
Tested locally:
image

**Comment by hi-rustin** [comment]
@pracucci Could you please take a look? Thanks! 💚 💙 💜 💛 ❤️

File: pkg/api/handlers_test.go [hunk 2]
```diff
@@ -235,3 +235,21 @@ func TestStatusConfigHandler(t *testing.T) {
...
+	cfg.Flags["flag1"] = "value1"
+	cfg.Flags["flag2"] = "value2"
+	h := cfg.statusFlagsHandler()
+	h(w, req)
+	resp := w.Result()
+	assert.Equal(t, 200, resp.StatusCode)
+
+	body, err := io.ReadAll(resp.Body)
+	assert.NoError(t, err)
+	assert.Equal(t, "{\"status\":\"success\",\"data\":{\"flag1\":\"value1\",\"flag2\":\"value2\"}}", string(body))
```
Comment (dimitarvdimitrov) [inline pkg/api/handlers_test.go:254]: nit: you can use `JSONEq` and use backticks for the json string so it's slightly easier to read?
```suggestion
	assert.JSONEq(t, `{"status":"success","data":{"flag1":"value1","flag2":"value2"}}`, string(body))
```

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM, nicely done :)

I have only one nitpick. I'll let someone else take a look too

**Comment by 56quarters** [comment]
This change looks good but I'm a bit worried that we're exposing CLI flags under a similar path as the rest of the Prometheus-compatible API. In the helm chart, anything under `/prometheus` not specifically sent to another component gets routed to the query-frontend. Putting secrets in CLI arguments isn't a great idea but I think exposing them via API under a path that would typically be public-facing is surprising behavior.

The original motivation for this change was to let Kiali detect that it's talking to something Prometheus-compatible. Maybe it makes sense to return an HTTP 200 with an empty `data` field. Maybe it even makes sense to not return any data from the `/api/v1/status/config` endpoint added in #3596?  I'm curious what other people think about this.

**Comment by 56quarters** [comment]
Hey @hi-rustin, We've decided to implement "no-op" versions of these endpoints that return empty configuration instead of the actual Mimir configuration. Thanks for your work on this but I'm going to go ahead and close this PR.

**Pull request closed without merging**